Content.Shared/Doors/Components/FirelockComponent.cs

```diff
@@ -20,24 +20,37 @@ public sealed class FirelockComponent : Component
         [DataField("autocloseDelay")] public TimeSpan AutocloseDelay = TimeSpan.FromSeconds(3f);
 
         /// <summary>
-        /// Maximum pressure difference before the firelock will refuse to open, in kPa.
+        ///   Maximum pressure difference before the firelock will refuse to open, in kPa.
         /// </summary>
         [DataField("pressureThreshold"), ViewVariables(VVAccess.ReadWrite)]
         public float PressureThreshold = 20;
 
         /// <summary>
-        /// Maximum temperature difference before the firelock will refuse to open, in k.
+        ///   Maximum temperature difference before the firelock will refuse to open, in k.
         /// </summary>
         [DataField("temperatureThreshold"), ViewVariables(VVAccess.ReadWrite)]
         public float TemperatureThreshold = 330;
         // this used to check for hot-spots, but because accessing that data is a a mess this now just checks
         // temperature. This does mean a cold room will trigger hot-air pop-ups
 
         /// <summary>
-        /// If true, and if this door has an <see cref="AtmosAlarmableComponent"/>, then it will only auto-close if the
-        /// alarm is set to danger.
+        ///   If true, and if this door has an <see cref="AtmosAlarmableComponent"/>, then it will only auto-close if the
+        ///   alarm is set to danger.
         /// </summary>
         [DataField("alarmAutoClose"), ViewVariables(VVAccess.ReadWrite)]
         public bool AlarmAutoClose = true;
+
+        /// <summary>
+        ///   Dangerous Fire was detected behind this door on the last check
+        /// </summary>
+        [DataField("dangerFire")]
+        public bool DangerFire = false;
+
+        /// <summary>
+        ///   Dangerous Pressure was detected behind this door on the last check
+        /// </summary>
+        [DataField("dangerPressure")]
+        public bool DangerPressure = false;
+
```

Comment:
This is unrelated to the component itself and is only used for GD so shouldn't be here.

Reply:
Removing these has performance costs (need a call to `CheckPressureAndFire` when DoorMetric runs), Not sure if we should save these somewhere else, or just pay to recalculate.